app/components/sendToken/index.js

```diff
@@ -221,6 +223,71 @@ export const Manual = props => {
     await checkLiquidityPair();
   }, [selectedToken, selectedToToken, path]);
 
+  useEffect(async () => {
+    calculatePriceImpact();
+  }, [amountIn, fromAmount]);
+
+  const calculatePriceImpact = async () => {
+    const rout = await router();
+    if (routeAddress.length === 2) {
+      const price = await rout.getAmountsOut(
+        '1000000000000000000',
+        routeAddress,
+      );
+
+      const marketPrice = ethers.utils.formatEther(price[1].toString());
+      const swapPrice = amountIn / fromAmount;
+      const priceDifference = swapPrice - marketPrice;
+      const priceImpact = (priceDifference / marketPrice) * 100;
+      setPriceImpact(parseFloat(priceImpact).toFixed(2));
+    } else if (routeAddress.length === 3) {
+      const price1 = await rout.getAmountsOut('1000000000000000000', [
+        routeAddress[0],
+        routeAddress[1],
+      ]);
+
+      const price1String = await price1.toString().split(',');
+
+      const price2 = await rout.getAmountsOut(price1String[1], [
+        routeAddress[1],
+        routeAddress[2],
+      ]);
+
+      const marketPrice = ethers.utils.formatEther(price2[1].toString());
+      const swapPrice = amountIn / fromAmount;
+      const priceDifference = swapPrice - marketPrice;
+      const priceImpact = (priceDifference / marketPrice) * 100;
+      setPriceImpact(parseFloat(priceImpact).toFixed(2));
+    } else if (routeAddress.length === 4) {
+      const price1 = await rout.getAmountsOut('1000000000000000000', [
+        routeAddress[0],
+        routeAddress[1],
+      ]);
+
+      const price1String = await price1.toString().split(',');
+
+      const price2 = await rout.getAmountsOut(price1String[1], [
+        routeAddress[1],
+        routeAddress[2],
+      ]);
+
+      const price2String = await price2.toString().split(',');
```

Comment:
toString split functions are not promise based so they dont need an await keyword.